fang/src/asynk/async_queue.rs

```diff
@@ -156,375 +136,289 @@ pub trait AsyncQueueable: Send {
 ///
 
 #[derive(TypedBuilder, Debug, Clone)]
-pub struct AsyncQueue<Tls>
-where
-    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
-    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
-    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
-    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
-{
+pub struct AsyncQueue {
     #[builder(default=None, setter(skip))]
-    pool: Option<Pool<PostgresConnectionManager<Tls>>>,
+    pool: Option<AnyPool>,
     #[builder(setter(into))]
     uri: String,
     #[builder(setter(into))]
     max_pool_size: u32,
     #[builder(default = false, setter(skip))]
     connected: bool,
+    #[builder(default = BackendSqlX::NoBackend, setter(skip))]
+    backend: BackendSqlX,
 }
 
 #[cfg(test)]
 use tokio::sync::Mutex;
 
 #[cfg(test)]
-static ASYNC_QUEUE_DB_TEST_COUNTER: Mutex<u32> = Mutex::const_new(0);
+static ASYNC_QUEUE_POSTGRES_TEST_COUNTER: Mutex<u32> = Mutex::const_new(0);
+
+#[cfg(test)]
+static ASYNC_QUEUE_SQLITE_TEST_COUNTER: Mutex<u32> = Mutex::const_new(0);
+
+#[cfg(test)]
+static ASYNC_QUEUE_MYSQL_TEST_COUNTER: Mutex<u32> = Mutex::const_new(0);
+
+#[cfg(test)]
+use sqlx::Executor;
+
+#[cfg(test)]
+use std::path::Path;
 
 #[cfg(test)]
-impl AsyncQueue<NoTls> {
+use std::env;
+
+use super::backend_sqlx::BackendSqlX;
+
+#[cfg(test)]
+impl AsyncQueue {
```

Comment:
can tests be moved to the end of the file?

Reply:
these are not tests, this `impl` is the way we construct a Queue for the tests, but i can move it to the end of the file

Reply:
done in this commit [adress comments and fix workflow](https://github.com/ayrat555/fang/pull/141/commits/192722db440e62c193614166febc6efe1bb1dadd)